src/Custom/Common/Instrumentation/InstrumentationFactory.cs

```diff
@@ -0,0 +1,60 @@
+﻿using OpenAI.Chat;
+using System;
+
+namespace OpenAI.Custom.Common.Instrumentation;
+
+internal class InstrumentationFactory
```

Comment:
@lmolkova, do you have a sense of whether our plan for generating third-party clients will be to generate these factory and scope types internally in the client?  Or are we planning to provide types to support this in System.ClientModel, similar to the ones in Azure.Core today?

Reply:
I think in the very long-term we will need a way to generate at least some of instrumentation code. But:
- it's very specific to the SDK and we'd need some support for it in typespec
- it's unlikely that we'll need anything in `System.ClientModel` - the Azure.Core stuff existed to bridge different generation of tracing API in .NET. Currently there is just one relevant set of APIs which client libraries can and should use directly.